src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,46 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplyTotal = 0;
+        int buyTotal = 0;
+
+        String line;
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                if (parts.length == 2) {
+                    String operation = parts[0].trim();
+                    int amount = Integer.parseInt(parts[1].trim());
+                    if ("supply".equalsIgnoreCase(operation)) {
+                        supplyTotal += amount;
+                    } else if ("buy".equalsIgnoreCase(operation)) {
+                        buyTotal += amount;
+                    }
+                }
+            }
+        } catch (NumberFormatException | IOException e) {
+            System.err.println("Error reading file: " + ((Exception)e).getMessage());
+            return;
```

This catch block violates checklist item #12: "Don't ignore exceptions Leaving an empty catch block or `e.printStackTrace` here is a bad practice. Better re-throw `RuntimeException` with the original exception and some message in the parameters:" 

Currently the code prints to `System.err` and returns, which hides the error and breaks proper error propagation. Replace this with rethrowing a RuntimeException that includes the original exception as the cause, e.g. `throw new RuntimeException("Can't read data from file " + fromFileName, e);`.